src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonArchitectureCheck.java

```diff
@@ -257,6 +264,16 @@ protected ClassTree getClassTree(JavaFileScannerContext fileContext) {
     return null;
   }
 
+  /**
+   * Creates a new {@link File} out of the currently analyzed file.
+   *
+   * @return {@link File} of the file to scan
+   */
+  protected File getFileToScan() {
+
+    return new File(this.context.getInputFile().toString());

```

Comment:
NPE in Travis seems to indicate that getInputFile() returns `null` in some cases.